src/pybind/mgr/dashboard_v2/tests/test_auth.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+
+import time
+from cherrypy.lib.sessions import RamSession
+from cherrypy.test import helper
+from mock import patch
+
+from ..auth import Auth
+from ..module import Module, cherrypy
+
+class AuthTest(helper.CPWebCase):
+    @staticmethod
+    def setup_server():
+        module = Module('dashboard', None, None)
+        auth = Auth(module)
+        cherrypy.tools.autenticate = cherrypy.Tool('before_handler', auth.check_auth)
+        cherrypy.tree.mount(auth, "/api/auth",
+                            config={'/ping': {'tools.autenticate.on': True}})
+        module.set_localized_config('session-expire','2')
+        module.set_localized_config('username','admin')
+        module.set_localized_config('password',
+            '$2b$12$KunrLI/uq7pqjvwUcAhIZu.B1dAGZ3liB8KFIJUOqZC.5/bEEmBQG')
+
+    def test_login_valid(self):
+        sess_mock = RamSession()
+        with patch('cherrypy.session', sess_mock, create=True):
+            self.getPage("/api/auth/login",
+                         body="username=admin&password=admin",
+                         method='POST')
+        self.assertStatus('200 OK')
+        self.assertBody('{"username": "admin"}')
+        self.assertEquals(sess_mock.get(Auth.SESSION_KEY), 'admin')
+
+    def test_login_invalid(self):
+        sess_mock = RamSession()
+        with patch('cherrypy.session', sess_mock, create=True):
+            self.getPage("/api/auth/login",
+                         body="username=admin&password=invalid",
+                         method='POST')
+        self.assertStatus('403 Forbidden')
+        self.assertBody('{"detail": "Invalid credentials"}')
+
+    def test_logout(self):
+        sess_mock = RamSession()
+        with patch('cherrypy.session', sess_mock, create=True):
+            self.getPage("/api/auth/login",
+                         body="username=admin&password=admin",
+                         method='POST')
+            self.assertEquals(sess_mock.get(Auth.SESSION_KEY), 'admin')
+            self.getPage("/api/auth/logout", method='POST')
+        self.assertStatus('200 OK')
```

Comment:
Why are the following lines outside the `with` block scope?

Reply:
Fixed